src/document.rs

```diff
@@ -1,75 +1,255 @@
-///     >>> doc.add_text(body, ("He was an old man who fished alone in a "
+///     >>> doc.add_text("title", "The Old Man and the Sea")
+///     >>> doc.add_text("body", ("He was an old man who fished alone in a "
 ///                             "skiff in the Gulf Stream and he had gone "
 ///                             "eighty-four days now without taking a fish."))
+///
+/// For simplicity, it is also possible to build a `Document` by passing the field
+/// values directly as constructor arguments.
+///
+/// Example:
+///     >>> doc = tantivy.Document(title=["The Old Man and the Sea"], body=["..."])
+///
+/// As syntactic sugar, tantivy also allows the user to pass a single values
+/// if there is only one. In other words, the following is also legal.
+///
+/// Example:
+///     >>> doc = tantivy.Document(title="The Old Man and the Sea", body="...")
+
 #[pyclass]
+#[derive(Default)]
 pub(crate) struct Document {
-    pub(crate) inner: tv::Document,
+    pub(crate) field_values: BTreeMap<String, Vec<tv::schema::Value>>,
+}
+
+impl fmt::Debug for Document {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let doc_str = self
+            .field_values
+            .iter()
+            .map(|(field_name, field_values)| {
+                let mut values_str =
+                    field_values.iter().map(value_to_string).join(",");
+                values_str.truncate(10);
+                format!("{}=[{}]", field_name, values_str)
+            })
+            .join(",");
+        write!(f, "Document({})", doc_str)
+    }
+}
+
+fn add_value<T>(doc: &mut Document, field_name: String, value: T)
+where
+    Value: From<T>,
+{
+    doc.field_values
+        .entry(field_name)
+        .or_insert_with(Vec::new)
+        .push(Value::from(value));
+}
+
+fn extract_value(any: &PyAny) -> PyResult<Value> {
+    if let Ok(s) = any.extract::<String>() {
+        return Ok(Value::Str(s));
+    }
+    if let Ok(num) = any.extract::<i64>() {
+        return Ok(Value::I64(num));
+    }
+    if let Ok(num) = any.extract::<f64>() {
+        return Ok(Value::F64(num));
+    }
+    if let Ok(py_datetime) = any.downcast_ref::<PyDateTime>() {
+        let datetime = Utc
+            .ymd(
+                py_datetime.get_year(),
+                py_datetime.get_month().into(),
+                py_datetime.get_day().into(),
+            )
+            .and_hms_micro(
+                py_datetime.get_hour().into(),
+                py_datetime.get_minute().into(),
+                py_datetime.get_second().into(),
+                py_datetime.get_microsecond(),
+            );
+        return Ok(Value::Date(datetime));
+    }
+    if let Ok(facet) = any.downcast_ref::<Facet>() {
+        return Ok(Value::Facet(facet.inner.clone()));
+    }
+    Err(to_pyerr(format!("Value unsupported {:?}", any)))
+}
+
+fn extract_value_single_or_list(any: &PyAny) -> PyResult<Vec<Value>> {
+    if let Ok(values) = any.downcast_ref::<PyList>() {
```

Comment:
Facet values fail on casting if they are not already Facet objects

Reply:
To be clear, you mean we need this to work if they are strings too correct?

Reply:
yups, its easier from API point of view to not needing to cast as facet objects, imho.

Reply:
This seem like a good idea. Can you open an issue and double check with @poljar we want this?